var/spack/repos/builtin/packages/glvis/package.py

```diff
@@ -0,0 +1,148 @@
+##############################################################################
+from spack import *
+
+
+class Glvis(MakefilePackage):
+    """GLVis: an OpenGL tool for visualization of FEM meshes and functions"""
+
+    homepage = "http://glvis.org/"
+    url      = "https://github.com/GLvis/glvis/archive/v3.3.tar.gz"
+
+    # glvis (like mfem) is downloaded from a URL shortener at request
+    # of upstream author Tzanio Kolev <tzanio@llnl.gov>.  See here:
+    # https://github.com/mfem/mfem/issues/53
+    #
+    # The following procedure should be used to verify security when a
+    # new version is added:
+    #
+    # 1. Verify that no checksums on old versions have changed.
+    #
+    # 2. Verify that the shortened URL for the new version is listed at:
+    #    http://glvis.org/download/
+    #
+    # 3. Use http://getlinkinfo.com or similar to verify that the
+    #    underling download link for the latest version comes has the
+    #    prefix: http://glvis.github.io/releases
+    #
+    # If this quick verification procedure fails, additional discussion
+    # will be required to verify the new version.
+
+    version('develop', git='https://github.com/glvis/glvis', branch='master')
+
+    version('3.3',
+            '1201a76d1b0c38240186c06f6478de77',
+            url='http://goo.gl/C0Oadw',
+            extension='.tar.gz')
+
+    version('3.2',
+            '4575a03a50d4730c07b5d4018e936707',
+            url='http://goo.gl/hzupg1',
+            extension='.tar.gz')
+
+    version('3.1',
+            '3ce8a53209c6593e066e568dbbf2bbf6',
+            url='http://goo.gl/gQZuu9',
+            extension='tar.gz')
+
+    variant('screenshots',
+            default='png',
+            values=('xwd', 'png', 'tiff'),
+            description='Backend used for screenshots')
+    variant('fonts', default=True,
+            description='Use antialiased fonts via freetype & fontconfig')
+
+    depends_on('mfem@develop', when='@develop')
+    depends_on('mfem@3.3', when='@3.3')
+    depends_on('mfem@3.2', when='@3.2')
+    depends_on('mfem@3.1', when='@3.1')
+
+    depends_on('gl')
+    depends_on('glu')
+    depends_on('libx11')
+
+    depends_on('libpng', when='screenshots=png')
+    depends_on('libtiff', when='screenshots=tiff')
+    depends_on('freetype', when='+fonts')
+    depends_on('fontconfig', when='+fonts')
+
+    def edit(self, spec, prefix):
+
+        def yes_no(s):
+            return 'YES' if self.spec.satisfies(s) else 'NO'
+
+        mfem = spec['mfem']
+        config_mk = mfem.package.config_mk
+
+        pkgs_libs = [('glu', 'libGLU'), ('gl', 'libGL'), ('libx11', 'libX11')]
+        search_dirs = ['lib64', 'lib']
+        gl_libs = LibraryList([])
+        for pkg, lib_ in pkgs_libs:
+            for dir in search_dirs:
+                lib = find_libraries([lib_], join_path(spec[pkg].prefix, dir),
+                                     shared=True, recursive=False)
```

Comment:
One of the reasons we added `spec.libs` and `spec.headers` is because Spack becomes really clunky when every package that depends on `gl` or `libx11` has to figure out where the libraries reside and what they are called. You end up with duplicated work, and as soon as the dependency package changes, things rapidly become out of sync. I would override `libs` (and `headers` if you want to include special directories/files) for each of these dependencies so you don't have to duplicate the work.

Reply:
Done.